10up/insert-special-characters
# fix: copy the current formats when insertings a character

### Description of the Change
Previously we would just insert the character at the point, however, when a character was inserted within a text format they would split the format and insert the character with no formats.

With this commit we copy the current `value` and insert out character here which preserves the surrounding formatting

Old Behaviour

https://user-images.githubusercontent.com/7556775/221526499-2ab9ff1d-e0be-4620-aa93-297cb53f9ebc.mov



New Behaviour

https://user-images.githubusercontent.com/7556775/221526534-449a7639-f591-425c-a666-c8b4bd56be6e.mov


### How to test the Change

Create some text with a text format (bold/italic) put cursor in the middle add a special character ensure that the format isnt split by the special character

### Changelog Entry
> Fixed - Maintain text formatting when inserting a character.


### Credits
@jrmd


### Checklist:
